pio-parser/src/lib.rs

```diff
@@ -213,10 +213,11 @@ impl<'a> ProgramState<'a> {
     }
 
     fn resolve(&self, name: &str) -> i32 {
-        match self.defines.get(name) {
-            Some(v) => v.1,
-            None => self.file_state.defines[name].1,
-        }
+        self.defines
+            .get(name)
+            .or_else(|| self.file_state.defines.get(name))
+            .unwrap_or_else(|| panic!("Unknown label {}", name))
```

Comment:
I think this should Err instead.
Panics aren't great in proc-macros as it prevents the generation of useful diagnostic messages.

Reply:
Sure, an error message pointing to the actual line of PIO code where the unknown label was used would be much better.

But just returning an `Err` from `resolve` wouldn't do that, the error must also be passed up to the macro, and properly translated to an error message there.

Current error message is:
```
thread 'test_unknown_label' panicked at 'no entry found for key', pio-parser/src/lib.rs:218:21
```

With this merge request, it becomes:
```
thread 'test_unknown_label' panicked at 'Unknown label some_unknown_label', pio-parser/src/lib.rs:219:32
```

So the message now says what went wrong. I'd say that's a huge improvement, even if the line numbers still point to the parsing code, not to actual error location.

As #30 has been open for half a year without anyone finding some time to fix it, I'd prefer to merge the sub-optimal solution for now instead of waiting much longer for a better solution.